Applications/Spire/Source/Ui/IntegerInputWidget.cpp

```diff
@@ -8,82 +8,31 @@
 using namespace boost::signals2;
 using namespace Spire;
 
-namespace {
-  auto BUTTON_X() {
-    static auto x = scale_width(12);
-    return x;
-  }
+IntegerInputWidget::IntegerInputWidget(int value, QWidget* parent)
```

Comment:
What do you think about using composition instead of inheritance for this?

Reply:
Wouldn't it have the same problem if it was using a DecimalInputWidget?

Reply:
Can you rephrase this question?

Reply:
I'm guessing based on you asking about using composition versus inheritance that there is a way to reuse a common numeric input for each of the numeric types. I'm not sure how I can modify the default QSpinBox implementations to create a generic numeric input that can be reused.

I can create a numeric input that uses Quantity as the base type that can be reused (assuming Quantity can express all our required numeric types), but that would require redoing some work regarding editing and fixing up the text in the widget.

Reply:
```QDoubleSpinBox``` can be used for all the widgets using composition instead of inheritance.

Reply:
You can inherit from ```QDoubleSpinBox``` to implement the ```DecimalInputWidget``` as you have done, and then reuse it for the ```QuantityInputWidget``` and ```IntegerInputWidget```.

Reply:
I don't understand how that's different from inheritance in this case. The Quantity input for instance would still have to convert a double to a quantity (and vice-versa) if it was using the DecimalInputWidget.

Reply:
No, you can not operate on the ```double```, you would need to operate on the ```QString``` to parse it into a ```Quantity```.

Reply:
https://github.com/spiretrading/nexus/blob/3151c02be77001d09b15b2fcc5a30d8572328f12/Nexus/Include/Nexus/Definitions/Quantity.hpp#L37

Reply:
Similarly when you implement the ```MoneyInputWidget``` in the future, you will also parse it into a ```Money```.

Reply:
In order to call setMinimum(double), would I have to do something like:

`
auto qty = Quantity(1.234567)
auto str = item_delegate.displayText(qty);
setMinimum(str.toDouble());
`

Reply:
Actually, displayText just casts the Quantity to a double anyway.